llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -446,6 +446,14 @@ class DXILBindingMap {
     return Pos == CallMap.end() ? Infos.end() : (Infos.begin() + Pos->second);
   }
 
+  /// Resolves a resource handle into a vector of ResourceBindingInfos that
+  /// represent the possible unique creations of the handle. Certain cases are
+  /// ambiguous so mulitple creation instructions may be returned. The resulting
+  /// ResourceBindingInfo can be used to depuplicate unique handles that
+  /// reference the same resource
+  SmallVector<dxil::ResourceBindingInfo>
+  findCreationInfo(const Value *Key) const;
```

Comment:
I'm not entirely convinced this name is clearer than the findByValue type name in the original, though I do agree that name is a bit ambiguous as well. I guess we want to convey something like "this looks up the possible resources this could be" - maybe "walkDefinitions" or something could work? In any case I'll leave it to you.

Reply:
Since I need to fix the typo anyways I'll just go back to the original name and keep it in the back of my mind for improvement. We still have to touch the function at least one more time so there will be another opportunity to review the name